### PR TITLE
REF: implement is_exact_match

### PR DESCRIPTION
**Pull request by jbrockmendel**
Working to smooth out some inconsistencies in setitem between code paths.

### CODE REVIEW

File: pandas/core/indexers.py [hunk 1]
```diff
@@ -270,6 +270,27 @@ def maybe_convert_indices(indices, n: int):
 # Unsorted
 
 
+def is_exact_match(target: ArrayLike, value: ArrayLike) -> bool:
```
Comment (jreback) [inline pandas/core/indexers.py:273]: maybe is_exact_shape_match?

File: pandas/core/indexers.py [hunk 2]
```diff
@@ -270,6 +270,27 @@ def maybe_convert_indices(indices, n: int):
...
+    ----------
+    target : np.ndarray or ExtensionArray
+    value : np.ndarray or ExtensionArray
+
+    Returns
+    -------
+    bool
+    """
+    return (
+        len(value.shape) > 0
```
Comment (jreback) [inline pandas/core/indexers.py:287]: `> 0` is redundant

Comment (jbrockmendel) [reply pandas/core/indexers.py:287]: needed for mypy

File: pandas/core/indexers.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexers.py:273]: sure

**Comment by jbrockmendel** [comment]
updated + greenish

**Pull request merged**